build.gradle.kts

```diff
@@ -62,164 +36,38 @@ dependencies {
-    showOnlySlow = false
-    showStandardStreams = false
-    showPassedStandardStreams = true
-    showSkippedStandardStreams = true
-    showFailedStandardStreams = true
-    logLevel = LogLevel.LIFECYCLE
-}
-
-// Configure Gradle Changelog Plugin - read more: https://github.com/JetBrains/gradle-changelog-plugin
-changelog {
-    groups.empty()
-    repositoryUrl = properties("pluginRepositoryUrl")
-}
-
-tasks.withType<Test> {
-    environment("GRADLE_RELEASE_REPOSITORY","https://services.gradle.org/distributions")
-    systemProperty("idea.log.leaked.projects.in.tests", "false")
-    systemProperty( "idea.maven.test.mirror", "https://repo1.maven.org/maven2")
-    systemProperty( "com.redhat.devtools.intellij.telemetry.mode", "disabled")
-}
-
-tasks.withType<Copy> {
-    duplicatesStrategy = DuplicatesStrategy.INCLUDE
-}
-
-tasks.register<Test>("integrationTest") {
-    useJUnitPlatform()
-    description = "Runs the integration tests."
-    group = "verification"
-    testClassesDirs = sourceSets["integrationTest"].output.classesDirs
-    classpath = sourceSets["integrationTest"].runtimeClasspath
-    outputs.upToDateWhen { false }
-    mustRunAfter(tasks["test"])
-}
-
-// Configure Gradle Kover Plugin - read more: https://github.com/Kotlin/kotlinx-kover#configuration
-koverReport {
-    defaults {
-        xml {
-            onCheck = true
-        }
-    }
+    version.set("2023.3")
+    type.set("IC") // Target IDE Platform
+    plugins.set(listOf("com.redhat.devtools.intellij.telemetry:1.2.1.62", "textmate", "properties"))
 }
 
 tasks {
-    wrapper {
-        gradleVersion = properties("gradleVersion").get()
+    // Set the JVM compatibility versions
+    withType<JavaCompile> {
+        sourceCompatibility = "17"
+        targetCompatibility = "17"
     }
-
-    runPluginVerifier {
-        failureLevel = listOf(INVALID_PLUGIN, COMPATIBILITY_PROBLEMS, MISSING_DEPENDENCIES )
-        verificationReportsFormats = listOf(MARKDOWN, HTML)
+    withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile> {
+        kotlinOptions.jvmTarget = "17"
     }
 
     patchPluginXml {
-        version = properties("pluginVersion")
-        sinceBuild = properties("pluginSinceBuild")
-        untilBuild = properties("pluginUntilBuild")
-        //TODO inject changelog into plugin.xml change-notes
-    }
-
-    // TODO: Once LSP4IJ no longer supports 2023.*, the hotswap JVM args can be added unconditionally
-    fun supportsEnhancedClassRedefinition(): Boolean {
-        val platformVersion = properties("platformVersion").getOrNull()
-        return when (platformVersion) {
-            // Versions before 2024.1 don't support enhanced class redefinition
-            "2023.2", "2023.3" -> false
-            else -> true
-        }
-    }
-
-    runIde {
-        // Improved hotswap for the IDE's JVM
-        if (supportsEnhancedClassRedefinition()) {
-            jvmArgs("-XX:+AllowEnhancedClassRedefinition", "-XX:HotswapAgent=fatjar")
```

Comment:
I haven't checked whether or not the Gradle 2.x `runIde` task is already including these args. Assuming it's not, we'll need to figure out how to retain them conditionally based on the executed IDE version.